src/line.rs

```diff
@@ -0,0 +1,58 @@
+/// A macro for creating a [`Line`] using vec! syntax.
+///
+/// `line!` is similar to the [`vec!`] macro, but it returns a [`Line`] instead of a `Vec`.
+///
+/// # Examples
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::{line, raw};
+///
+/// let line = line!["hello", "world"];
+/// let line = line![raw!("hello {}", "world"), raw!("goodbye {}", "world")];
+/// let line = line!["hello".red(), "world".red().bold()];
+/// let line = line!["hello"; 2];
+/// ```
+///
+/// [`Line`]: crate::text::Line
+#[macro_export]
+macro_rules! line {
+    () => {
+        ratatui::text::Line::default()
+    };
+    ($span:expr; $n:expr) => {
+      ratatui::text::Line::from(vec![$span.into(); $n])
+    };
+    ($($span:expr),+ $(,)?) => {{
+        ratatui::text::Line::from(vec![
+        $(
+            $span.into(),
+        )+
+        ])
+    }};
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::raw;
+    use ratatui::prelude::*;
+
+    #[test]
+    fn line() {
+        // literal
+        let line = line!["hello", "world"];
+        assert_eq!(line, Line::from(vec!["hello".into(), "world".into()]));
+
+        // raw instead line
+        let line = line![raw!("hello"), "world"];
+        assert_eq!(line, Line::from(vec!["hello".into(), "world".into()]));
+
+        // vec count syntax
+        let line = line!["hello"; 2];
```

Comment:
Maybe add this and another easy to read for ratatui users to the doc comment as examples so it can be understood better from reading the docs?

Reply:
This was already in the examples, but I'm reformatted the examples to make it more aligned with the `vec!` documentation.